packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -283,7 +283,10 @@ class _ButtonStyleState extends State<ButtonStyleButton> with TickerProviderStat
     final Size? resolvedMaximumSize = resolve<Size?>((ButtonStyle? style) => style?.maximumSize);
     final double? resolvedIconSize = resolve<double?>((ButtonStyle? style) => style?.iconSize);
     final BorderSide? resolvedSide = resolve<BorderSide?>((ButtonStyle? style) => style?.side);
-    final OutlinedBorder? resolvedShape = resolve<OutlinedBorder?>((ButtonStyle? style) => style?.shape);
+    ShapeBorder? resolvedShape = resolve<ShapeBorder?>((ButtonStyle? style) => style?.shape);
+    if (resolvedShape is OutlinedBorder) {
+      resolvedShape = resolvedShape.copyWith(side: resolvedSide);
+    }
```

Comment:
This is odd. Why does OutlinedBorder need special casing? Is the side just dropped on the floor if shape is not an OutlineBorder?

Reply:
Actually, when I look at supporting `Border` and `BorderDirectional`, it doesn't really make sense, because basically all they are is a way to specify different border sides for each side, and if you specified one, and then the button's `side` overwrites them all, then it isn't much use, you might as well use a `RoundedRectangleBorder` instead.  So for `Border` and `BorderDirectional`, I say yes, the `side` just gets dropped on the floor.

Reply:
OK, I'm just going to revert this.  It's too weird.

It means that people will wonder about why menu panels can use a `Border`, but menu buttons cannot, but that's not as weird as this was getting.